test/expandKeyValue_test.js

```diff
@@ -1,56 +1,48 @@
-/**
- * Copyright 2012 Google Inc. All rights reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
+goog.require('helper');
 
-/**
- * @fileoverview Unit tests for expandKeyValue_.
- * @author bnkuhn@gmail.com (Brian Kuhn)
- */
-
-test('expandKeyValue_', function() {
-  function assertExpand(key, value, expected) {
-    deepEqual(helper.expandKeyValue_(key, value), expected);
+describe('The helper.expandKeyValue_ function', () => {
+  function assertExpand(key, value, expected, context = '') {
+    expect(helper.expandKeyValue_(key, value))
+        .withContext(context)
+        .toEqual(expected);
   }
 
-  assertExpand('a', 1, {a: 1});
-  assertExpand('a', 0, {a: 0});
-  assertExpand('a', -1, {a: -1});
-  assertExpand('a', null, {a: null});
-  assertExpand('a', undefined, {a: undefined});
-  assertExpand('a', true, {a: true});
-  assertExpand('a', false, {a: false});
-  assertExpand('a', [], {a: []});
-  assertExpand('a', ['b', {}, 3], {a: ['b', {}, 3]});
-  assertExpand('a', {b: {c: 3, d: 4}}, {a: {b: {c: 3, d: 4}}});
+  it('successfully converts key value pairs into objects when the key string is non dot-separated', () => {
```

Comment:
In case if we decide to test this on IE, we'll need to make sure that tranpilation is enabled
https://caniuse.com/#feat=arrow-functions

Reply:
This is going to be a tricky problem - we could theoretically try getting a babel preprocessor to run against our .js files but it will differ significantly from our compiled code.

We are using karma-googmodule-preprocessor to preprocess our files using closure compiler and unfortunately we do not have the ability to pass in compilation flags like LANGUAGE_OUT.

I've reverted to using function() syntax for the time being.

If testing on IE is a must have, we could possibly test against our distributed files (similar to PR #22) but that wouldn't exactly be unit testing anymore.